hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java

```diff
@@ -27,19 +29,49 @@
  */
 public class ApiStatusEnricher extends AbstractTraceEnricher {
 
+  public static String getStatusCode(Event event, Protocol protocol) {
+    List<String> statusCodeKeys = Lists.newArrayList();
+    if (Protocol.PROTOCOL_GRPC == protocol) {
+      if (event.getGrpc() != null && event.getGrpc().getResponse() != null) {
+        int statusCode = event.getGrpc().getResponse().getStatusCode();
+        // Checking for the default value for status code field
+        if (statusCode != -1) {
+          return Integer.toString(statusCode);
+        }
+      }
+      statusCodeKeys.add(RawSpanConstants.getValue(CensusResponse.CENSUS_RESPONSE_STATUS_CODE));
+      statusCodeKeys.add(RawSpanConstants.getValue(Grpc.GRPC_STATUS_CODE));
+      statusCodeKeys.add(RawSpanConstants.getValue(CensusResponse.CENSUS_RESPONSE_CENSUS_STATUS_CODE));
+    } else if (Protocol.PROTOCOL_HTTP == protocol || Protocol.PROTOCOL_HTTPS == protocol) {
+      statusCodeKeys.add(RawSpanConstants.getValue(OTSpanTag.OT_SPAN_TAG_HTTP_STATUS_CODE));
+      statusCodeKeys.add(RawSpanConstants.getValue(Http.HTTP_RESPONSE_STATUS_CODE));
+    }
+
+    return SpanAttributeUtils.getFirstAvailableStringAttribute(event, statusCodeKeys);
+  }
+
   @Override
   public void enrichEvent(StructuredTrace trace, Event event) {
     Protocol protocol = EnrichedSpanUtils.getProtocol(event);
     String statusCode = getStatusCode(event, protocol);
+
     String statusMessage = null;
     String status = null;
     if (Protocol.PROTOCOL_GRPC == protocol) {
       status = GrpcCodeMapper.getState(statusCode);
 
-      List<String> statusMessageKeys = Lists.newArrayList();
-      statusMessageKeys.add(RawSpanConstants.getValue(CensusResponse.CENSUS_RESPONSE_STATUS_MESSAGE));
-      statusMessageKeys.add(RawSpanConstants.getValue(Envoy.ENVOY_GRPC_STATUS_MESSAGE));
-      statusMessage = SpanAttributeUtils.getFirstAvailableStringAttribute(event, statusMessageKeys);
+      if (event.getGrpc() != null && event.getGrpc().getResponse() != null) {
+        if (StringUtils.isNotBlank(event.getGrpc().getResponse().getStatusMessage())) {
```

Comment:
I think once we are here it makes sense to define `Response response = event.getGrpc().getResponse()` (thanks Java) so we avoid chaining methods over and over and improve readability, other than that it is quite hard to read the method. Does it make sense?

Reply:
Done.